frontend/src/pages/projects/projectSharing/useGroups.ts

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import useFetchState, { FetchState } from '~/utilities/useFetchState';
+import { listGroups } from '~/api';
+import { GroupKind } from '~/k8sTypes';
+
+const useGroups = (): FetchState<GroupKind[]> => {
+  const getGroup = React.useCallback(
+    () =>
+      listGroups().catch((e) => {
+        if (e.statusObject?.code === 404) {
+          throw new Error('No groups found.');
+        }
+        throw e;
+      }),
+
+    [],
+  );
```

Comment:
Ok, i think this could be a great solution, it's still using the synchronicity of the rest of the calls so it won't feel lagged when turning the permission tab and we don't get to extract another context and add more logic.